decisions/0005-remixing-react-router.md

```diff
@@ -0,0 +1,202 @@
+
+### `fetcher.load()` participates in revalidations
+
+In Remix, if you have data on a page from a `fetcher.load()` and you submit a mutation - those fetchers don't revalidate so the data may now be stale if the mutation impacted it. We've changed this in `@remix-run/router` such that revalidation updates _all_ active loaded data including route loaders as well as active `fetcher.load()` calls. These can be opted out of using the normal `shouldRevalidate()` method
+
+**Backwards Compatibility**
+
+We categorize this as a bug fix since fetchers get stale in current Remix apps
+
+### `useTransition` renamed to `useNavigation`
+
+This was done for two reasons:
+
+- Avoid confusion with the [`useTransition`][react usetransition] hook in React 18
+- It's more semantically correct because a "navigation" is what you trigger as a result of `router.navigate()` or `useNavigate()`
+
+**Backwards Compatibility**
+
+We plan to export `useNavigation` in Remix and encourage folks to switch, but we will continue to include `useTransition` in a deprecated fashion
+
+### Navigations/Fetchers state structure changes
+
+**`useTransition().type` is removed**
+
+In Remix, the `useTransition` hook returned a Transition object which had a `state` property of `"idle" | "loading" | "submitting"`. It also had a `type` property which represented sort of "sub-states" such as `"normalLoad" | "actionReload" | "loaderRedirect"` etc. In React Router we chose to get rid of the `type` field for 2 reasons:
+
+1. In practice, we found that the _vast_ majority of the time all you needed to reference was the `state`
+2. For scenarios in which you really do need to distinguish, we are pretty sure that in all cases, you can deduce the `type` from `state`, current location (`useLocation`), next location (`useNavigation().location`), and submission info (`useNavigation().formData`).
+
+**`useTransition().submission` is flattened**
+
+Another area that changes is the `useTransition().submission` property was removed. We found that in practice folks never really needed the submission as a standalone thing, and instead always just cared about the `formMethod` or `formData`. So we flattened them onto the navigation, so `useNavigation()` will return an object of the format:
+
+```
+{
+  state: "idle" | "loading" | "submitting";
+  location: Location;
+  formMethod?: FormMethod;
+  formAction?: string;
+  formEncType?: FormEncType;
+  formData?: FormData;
+}
+```
+
+**Backwards Compatibility**
+
+We plan to remain backwards compatible here in Remix. Very likely we'll expose the `useNavigation` hook and encourage users to move to that. And then `useTransition` will remain in a deprecated state and it will cal `useNavigation` and then backfill the `type` and `submission` properties.
```

Comment:
```suggestion
We plan to remain backwards compatible here in Remix. Very likely we'll expose the `useNavigation` hook and encourage users to move to that. And then `useTransition` will remain in a deprecated state and it will call `useNavigation` and then backfill the `type` and `submission` properties.
```